jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/deadlines/notifications/impl/NotificationListenerManager.java

```diff
@@ -30,18 +32,59 @@
  *
  */
 public class NotificationListenerManager {
-    
+
+    public static final String KIE_LISTENER_EXCLUDE = "org.kie.jpbm.notification_listeners.exclude";
+
+    public static final String KIE_LISTENER_INCLUDE = "org.kie.jpbm.notification_listeners.include";
+
     private static final Logger logger = LoggerFactory.getLogger(NotificationListenerManager.class);
-    
+
     private static ServiceLoader<NotificationListener> listenersLoaded = ServiceLoader.load(NotificationListener.class);
-    
+
     private static NotificationListenerManager INSTANCE = new NotificationListenerManager();
-    
+
     private List<NotificationListener> listeners = new ArrayList<NotificationListener>();
-    
+
+    private List<String> excludeLists;
+
+    private List<String> includeLists;
+
+    public static List<String> propertyToList(String propertyName) {
+        String name = System.getProperty(propertyName);
+        if (name == null) {
+            return null;
+        }
+        return Arrays.asList(name.split(","));
+    }
+
     private NotificationListenerManager() {
+        init();
+    }
+
+    public void reset() {
+        init();
+    }
+
+    private void init() {
+        excludeLists = propertyToList(KIE_LISTENER_EXCLUDE);
+        includeLists = propertyToList(KIE_LISTENER_INCLUDE);
+
+        listeners.clear();
+        Predicate<String> predicate = createPredicate();
         for (NotificationListener listener : listenersLoaded) {
-            listeners.add(listener);
+            if (predicate.test(listener.getClass().getName())) {
+                listeners.add(listener);
+            }
+        }
+    }
+
+    private Predicate<String> createPredicate() {
+        if(includeLists != null) {
```

Comment:
Missing a blank after "if"
If `propertyToList` returns empty list, check `if (!includeLists.isEmpty())`

Reply:
ok with the if. The other one I already explain why not.